solutions/usaco-920.mdx

```diff
@@ -5,10 +5,64 @@ title: The Great Revegetation
 author: Sofia Yang
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_revegetate_silver_feb19.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_revegetate_silver_feb19.html)
 
 <LanguageSection>
 
+<CPPSection>
+```cpp
+// Source: Official Analysis (http://www.usaco.org/current/data/sol_revegetate_silver_feb19.html)
+#include <iostream>
+#include <fstream>
+#include <vector>
+using namespace std;
+ 
+const int MAX_N=1e5;
+int L[MAX_N+1];
+vector<int> S_nbrs[MAX_N+1], D_nbrs[MAX_N+1];
+bool impossible;
+ 
+void visit ( int x , int v )
+{
+  L[x] = v;
+  for (auto n : S_nbrs[x]) {
+    if (L[n] == 3-v) impossible = true; 
+    if (L[n] == 0) visit(n, v);
+  }
+  for (auto n : D_nbrs[x]) {
+    if (L[n] == v) impossible = true; 
+    if (L[n] == 0) visit(n, 3-v);
+  }  
+}
+ 
+int main()
+{
+  ifstream fin ("revegetate.in");
+  int N, M, answer;
+  fin >> N >> M;
+  for (int i=0; i<M; i++) {
```

Comment:
can you reformat the code so that there are spaces around all operators? for example, this line should be
```suggestion
  for (int i = 0; i < M; i++) {
```
but do that for everything else too

Reply:
Resolved